doc/layer2-forwarding-enhancements/SONiC Layer 2 Forwarding Enhancements HLD.md

```diff
@@ -0,0 +1,444 @@
+
+
+# 1 Requirement Overview
+## 1.1 Functional Requirements
+
+ 1. FDB Flush Support
+	 - FDB entries should be flushed per Port when the operational state of the port goes down. Only dynamic entries should be flushed.
+	 - FDB entries should be flushed per Port per VLAN when a port is removed from a VLAN. Both dynamic and static entries should be flushed. Static FDB configuration should not be removed.
+	 - FDB entries should be flushed per Port or per Port per VLAN when triggered by Layer 2 Protocol upon topology change. Only dynamic entries should be flushed.
+	 - FDB entries should be flushed per Portchannel when the admin or operational status of the Portchannel goes down.
+	 - FDB entry should be removed from FDB_TABLE in STATE_DB and ASIC_SB and internal orchagent FDB data structure and Hardware.
+	
+2. Handle MAC move event generated by hardware.
+	- Existing FDB entry should be replaced with one having new Port in ASIC_DB and STATE_DB when a MAC move event is received.
+	
+3. Configuration CLI for FDB aging time.
+	- The FDB aging time should be configurable in hardware to a desired value from CLI.
+	- There should be an option to disable aging by setting the aging time to 0.
+	- The default FDB aging time should be set to 0. User has option to set desired value in default config file.
+	
+4. Configuration CLI for Static FDB entry.
+	- A Static FDB entry should be configurable in hardware from CLI.
+	- Static FDB entry should be set in CONFIG_DB and APP_DB even if the port is not member of VLAN.
+	- Static FDB entry should be added in saved FDB in Orchagent if port is not member of VLAN.
+	- Static FDB entry should be added in saved FDB in Orchagent after the entry is deleted due to FDB flush when port is removed from VLAN.
+	- Static FDB entry should be added in STATE_DB.
+	
+5. Should have per Port, per VLAN and  per Port per VLAN FDB clear options in CLI command "sonic-clear fdb".
+
+6. VLAN range CLI support
+	- Should be able to create a range of VLANs in a single CLI command.
+	- Should be able to delete a range of VLANs in a single CLI command.
+	- Should be able to add a port to a range of VLANs in a single CLI command.
+	- Should be able to delete a port from a range of VLANs in a single CLI command.
+
+
+## 1.2 Configuration and Management Requirements
+- New CLI is added for configuring FDB aging time.
+- New CLI is added to display current FDB aging time.
+- New CLI is added for add/delete of Static FDB entry.
+- Existing CLI 'sonic-clear fdb' is extended to clear FDB per port or per VLAN or per Port per VLAN.
+- Existing CLI tree is extended to include support for keyword 'range' in each of the VLAN create and delete and VLAN member create and delete commands.
+
+
+## 1.3 Scalability Requirements
+- Up to 4094 VLAN will be supported.
+- VLAN range commands are invalidated for VLANs that fall outside the range 1 through 4094
+  (Aforementioned range should be configurable and valid).
```

Comment:
What are the scaling numbers with MAC moves and aging?

Reply:
MAC move and aging has no specific scaling number. It is same as total number of MAC supported, which varies across platforms.